src/analysis/utils/runtime-utils.ts

```diff
@@ -74,15 +75,15 @@ export type NormalizedComputeRegion = NominalType<string, 'ComputeRegion'>;
 // TODO: test when zone and region have types
 export const getNormalizedComputeRegion = (config: GoogleRuntimeConfig | AzureConfig): NormalizedComputeRegion => {
   if (isGceConfig(config) || isGceWithPdConfig(config)) {
-    return getRegionFromZone(config.zone).toUpperCase() as NormalizedComputeRegion;
+    return (getRegionFromZone(config.zone) || defaultComputeRegion).toUpperCase() as NormalizedComputeRegion;
   }
   if (isDataprocConfig(config)) {
-    return config.region.toUpperCase() as NormalizedComputeRegion;
+    return (config.region || defaultComputeRegion).toUpperCase() as NormalizedComputeRegion;
```

Comment:
same here; seems like `region` is nonoptional anyway

Reply:
This code takes a `RuntimeConfig` which can be one of 4 concrete types, two of which have zone and two of which have region, and normalizes it to a region for both cost calculation and display.

Reply:
yes, but each of the specific checks is inside of a typeguard conditional like `isDataprocConfig(config)`, and knows which type it's working with.